### PR TITLE
Adds extractors tap-ms-dynamics (singer-io)

### PR DESCRIPTION
**Pull request by pnadolny13**
Closes https://github.com/meltano/hub/issues/1635

### CODE REVIEW

File: _data/meltano/extractors/tap-dynamics/singer-io.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+- discover
+- state
+description: CRM Platform
+domain_url: 
+  https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/use-microsoft-dynamics-365-web-api
+executable: tap-ms-dynamics
+keywords:
+- api
+label: Dynamics
+logo_url: /assets/logos/extractors/ms-dynamics.png
```
Comment (edgarrmondragon) [inline _data/meltano/extractors/tap-dynamics/singer-io.yml:12]:
```suggestion
logo_url: /assets/logos/extractors/dynamics.png
```

**Review by tayloramurphy — APPROVED** [review]

**Comment by pnadolny13** [comment]
I'm actually not sure if this is supposed to be a new plugin page or under this existing one. I need to look into a more. @dineshssdn-867 suggests that its different in https://github.com/meltano/hub/issues/1635#issuecomment-1884886056.

**Comment by dineshssdn-867** [comment]
@sbalnojan can, you pls approve this PR ?

**Comment by pnadolny13** [comment]
@dineshssdn-867 I can merge it when its ready since we already got approvals. If this doesnt fit in the existing plugin page https://hub.meltano.com/extractors/tap-dynamics, can you help me understand what the new plugin page should reference? I'm having a hard time understanding the dynamics suite of tools. It looks like the current hub page is for the CRM tool.

I also noticed that https://hub.meltano.com/extractors/tap-dynamics--bytecodeio/ is likely the first iteration of the tap in this PR https://github.com/singer-io/tap-ms-dynamics. If we do end up creating a new plugin page we'll likely want to move that too.

We need:
- label
- description
- logo image

**Comment by dineshssdn-867** [comment]
**Logo:-** 
image

**Label:**
tap-ms-dynamics

**Description:**

_Overview:_
The tap-ms-dynamics connector stands as a dedicated link to Microsoft Dynamics 365, a robust CRM and ERP platform. Tailored for seamless integration, this tap is crafted to proficiently extract, explore, and maintain data states from Dynamics 365, facilitating its smooth integration into diverse data pipelines.

_Data Extraction:_
The primary mission of the tap-ms-dynamics is to extract comprehensive data from Dynamics 365. This encompasses the retrieval of records, entities, relationships, and metadata from the Dynamics 365 instance. The tap employs Dynamics 365 APIs, ensuring a dependable and secure extraction process.

_Discovery Process:_
Enabling a thorough comprehension of the Dynamics 365 data landscape, the tap includes a dynamic discovery process. This process involves the identification and cataloging of entities, relationships, and schema information within the Dynamics 365 instance. It dynamically adapts to changes in the data model over time.

_State Management:_
Central to the tap's functionality is its state management mechanism, vital for enhancing efficiency and accuracy in the data pipeline. By tracking the last extracted data, the tap ensures that only new or modified records are fetched during subsequent extractions, optimizing data transfer and reducing processing time.

_Incremental Data Loading:_
Complementing its state management capabilities, the tap supports incremental data loading. This feature allows the extraction of only changed or new data since the last extraction, significantly reducing the amount of data transferred and enhancing overall pipeline performance.

_Configuration Differentiators:_
Noteworthy is the absence of the extra requirement for Airbyte Dockerization in tap-ms-dynamics, setting it apart from other Airbyte variant taps. Additionally, while other Dynamics taps often lack proper authentication configuration, tap-ms-dynamics ensures a secure connection to Dynamics 365. Personal testing has affirmed the robustness of this connector, with results available upon request.

In essence, the tap-ms-dynamics connector emerges as a sophisticated tool for data extraction from Dynamics 365, distinguished by its efficiency, adaptability, and security features, making it a valuable asset for data integration scenarios.

**Pull request merged**